src/webui/www/tstool.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# TSTool - script for update qBittorrent WebUI translation files
+# Copyright (C) 2018  Vladimir Golovnev <glassez@yandex.ru>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# In addition, as a special exception, the copyright holders give permission to
+# link this program with the OpenSSL project's "OpenSSL" library (or with
+# modified versions of it that use the same license as the "OpenSSL" library),
+# and distribute the linked executables. You must obey the GNU General Public
+# License in all respects for all of the code used other than "OpenSSL".  If you
+# modify file(s), you may extend this exception to your version of the file(s),
+# but you are not obligated to do so. If you do not wish to do so, delete this
+# exception statement from your version.
+ 
+from sys import argv
+import argparse
+import copy
+import os
+from os.path import splitext, join, normpath
+import re
+import xml.etree.ElementTree as ET
+
+accepted_exts = (".js", ".html", ".css")
```

Comment:
I would expect a list, a bit strange to see a tuple is used.

Reply:
Also I think .css & .js can be excluded?

Reply:
I'm not python developer. I use it very rarely...
I only used tuple because it's immutable. But apparently it doesn't matter, and the list is more appropriate (from use cases POV).

Reply:
>Also I think .css & .js can be excluded?

Why?
.js files contain translatable strings even in builtin WebUI. And I think .css can contain it for some purposes in altrernative WebUI.

Reply:
>.js files contain translatable strings even in builtin WebUI. 

Now I see... 

>And I think .css can contain it for some purposes in alternative WebUI.

So you mean this script can be also used for alternative WebUI translations?
I would let this script cover only built-in WebUI translations, let alternative WebUI provide their own i18n solution.
IMO it is not really sane to have translation in css files, but who knows?

Reply:
>why should we limit the use of this simple method hjust insert QBT_TR and place appropriate .qm files into "translations" folder)?

Personally I don't want to put this burden on the qbt server when there are client side solutions (maybe like http://airbnb.io/polyglot.js/ ?).
IIRC this has been discussed before, I'll stop here, lets not delve into it more.

Reply:
>Personally I don't want to put this burden on the qbt server when there are client side solutions

I don't understand your POV. This logic is already here (on qbt server side). I just changed it a bit to don't use GUI translations. Or do you suggest to remove it at all and use client side translation for all possible web ui? IMO, a very strange offer in the absence of active web developers in our team.

@sledgehammer999, you should read this discussion when you'll review this since it contains different POVs on the overall problem.